tests/foreman/ui_airgun/test_host.py

```diff
@@ -354,6 +355,65 @@ def test_positive_create_host_with_parameters(session, module_global_params):
             _get_set_from_list_of_dict(values['parameters']['global_params']))
 
 
+@tier3
+def test_positive_create_with_config_group(session, module_host_template):
+    """Create new Host with config group assigned to it
+
+    :id: 8834011e-f920-43de-8f28-1f370b2b8d69
+
+    :expectedresults: Host is created and contains correct config group
+
+    :CaseLevel: System
+    """
+    config_group = entities.ConfigGroup().create()
+    with session:
+        host_name = create_fake_host(
+            session,
+            module_host_template,
+            extra_values={'puppet_classes.config_groups.assigned': [config_group.name]}
+        )
+        assert session.host.search(host_name)[0]['Name'] == host_name
+        values = session.host.read(host_name, widget_names='puppet_classes')
+        assert len(values['puppet_classes']['config_groups']['assigned']) == 1
+        assert values['puppet_classes']['config_groups']['assigned'][0] == config_group.name
+
+
+@tier3
+def test_positive_create_with_puppet_class(
+        session, module_host_template, module_org, module_loc):
+    """Create new Host with puppet class assigned to it
+
+    :id: d883f169-1105-435c-8422-a7160055734a
+
+    :expectedresults: Host is created and contains correct puppet class
+
+    :CaseLevel: System
+    """
+    pc_name = 'generic_1'
```

Comment:
Why this is static? Can't we use it  gen_string('alpha') here?

If I rerun this test again on the same satellite box, then it may fail as this will run on the same org? (considering regression verification). please correct me if I'm wrong.

Reply:
that is name of puppet module, not just simple word:
https://omaciel.fedorapeople.org/bagoftricks/system/releases/r/robottelo/

And nope, it will not fail as you always get another location and organization